### PR TITLE
Introduce storage lookup benchmarks

### PR DESCRIPTION
**Pull request by ljedrz**
Since future performance improvements are likely to be increasingly intricate, it's a good idea to introduce some benchmarks in order to ensure that they are beneficial.

For now, they were used just to double-check the positive impact of https://github.com/AleoHQ/snarkOS/pull/1546 and https://github.com/AleoHQ/snarkOS/pull/1548.

Future PRs will likely introduce some insertion and deletion benchmarks as well.

### CODE REVIEW

File: storage/benches/lookups.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+fn lookups(c: &mut Criterion) {
+    // Prepare the test ledger.
+    let temp_dir = tempfile::tempdir().expect("Failed to open temporary directory").into_path();
+    let ledger = LedgerState::open_writer::<RocksDB, _>(temp_dir).expect("Failed to initialize ledger");
+
+    // Read the test blocks.
+    // note: the `blocks_100` and `blocks_1000` files were generated on a testnet2 storage using `LedgerState::dump_blocks`.
+    let mut test_blocks = fs::read(format!("benches/blocks_{}", NUM_BLOCKS)).expect(&format!("Missing the test blocks file"));
+    let blocks: Vec<Block<Testnet2>> = bincode::deserialize(&mut test_blocks).expect("Failed to deserialize a block dump");
+    assert_eq!(blocks.len(), NUM_BLOCKS - 1);
```
Comment (niklaslong) [inline storage/benches/lookups.rs:37]: The dumps don't include genesis?

**Review by niklaslong — APPROVED** [review]
LGTM 👍

File: storage/benches/lookups.rs (hunk 1 above)
Comment (ljedrz) [reply storage/benches/lookups.rs:37]: No, because the `LedgerState` inserts it automatically; in other words, it's not needed.

Comment (niklaslong) [reply storage/benches/lookups.rs:37]: Gotcha! 👍

**Comment by ljedrz** [comment]
Rebased.

**Review by howardwu — APPROVED** [review]

**Pull request merged**